torch/utils/data/sampler.py

```diff
@@ -346,3 +347,45 @@ def __len__(self) -> int:
             return len(self.sampler) // self.batch_size  # type: ignore[arg-type]
         else:
             return (len(self.sampler) + self.batch_size - 1) // self.batch_size  # type: ignore[arg-type]
+
+class RandomBatchSampler(Sampler[list[int]]):
+    def __init__(
+            self,
+            data_source: Sized,
+            replacement: bool = False,
+            generator : np.random.Generator = None,
```

Comment:
for generator, it is usually, torch.Generator, no? any reason for NumPy generator here?